snippets/rust.snippets

```diff
@@ -19,32 +19,53 @@ snippet bench "Bench function" b
 			${0}
 		})
 	}
-snippet new
+snippet new "Constructor function"
 	pub fn new(${2}) -> ${1:Name} {
-		${0}return $1 { ${3} };
+		$1 { ${3} };
 	}
-snippet main
+snippet main "Main function"
 	pub fn main() {
 		${0}
 	}
-snippet let
-	let ${1:name: type} = ${2};
-snippet let
-	let mut ${1:name: type} = ${2};
-snippet pln
+snippet let "let variable declaration"
+	let ${1:name}${2:: ${3:type}} = ${4};
+snippet letm "let mut variable declaration"
+	let mut ${1:name}${2:: ${3:type}} = ${4};
```

Comment:
These are so commonly used that I think it's better to have two separate triggers than one with either two definitions or a tabstop for `mut` that needs to be deleted when unwanted.

Note that I only assume the nested tabstops work based on the Snipmate documentation, as noted in the main PR comment UltiSnips doesn't seem to support them in the Snipmate compatibility layer, so these are duplicated in the UltiSnips snippets.